config/config-server/test/com/thoughtworks/go/config/MagicalGoConfigXmlWriterTest.java

```diff
@@ -339,6 +339,49 @@ public void shouldWriteEmptyOnCancelTaskWhenDefined() throws Exception {
     }
 
     @Test
+    public void shouldNotWriteMoreThanOneOnCancelTaskWhenDefined() throws Exception {
+        String xml = "<cruise schemaVersion='30'>\n"
+                + "<server artifactsdir='artifactsDir' >"
+                + "</server>"
+                + "<pipelines>\n"
+                + "<pipeline name='pipeline1' template='abc'>\n"
+                + "    <materials>\n"
+                + "      <svn url ='svnurl' username='foo' password='password'/>"
+                + "    </materials>\n"
+                + "</pipeline>\n"
+                + "</pipelines>\n"
+                + "<templates>\n"
+                + "  <pipeline name='abc'>\n"
+                + "    <stage name='stage1'>"
+                + "      <jobs>"
+                + "        <job name='job1'>"
+                + "         <tasks>"
+                + "             <exec command=\"rake\">\n"
+                + "                 <arg>all_test</arg>\n"
+                + "                 <oncancel>\n"
+                + "                     <ant target='kill' />\n"
+                + "                     <ant target='kill' />\n"
+                + "                 </oncancel>\n"
+                + "             </exec>"
+                + "         </tasks>"
+                + "        </job>"
+                + "      </jobs>"
+                + "    </stage>"
+                + "  </pipeline>\n"
+                + "</templates>\n"
+                + "</cruise>";
+
+
+        try {
+            CruiseConfig cruiseConfig = ConfigMigrator.loadWithMigration(IOUtils.toInputStream(xml)).config;
+            xmlWriter.write(cruiseConfig, output, false);
+            fail("Should not allow user to write more than one oncancel task");
+        } catch (Exception e) {
+            assertThat(e.getMessage(), containsString("Invalid content was found starting with element 'ant'. No child element is expected at this point."));
```

Comment:
Can we improve this error message? @jyotisingh or @juhijariwala would know how :)